numpy/core/tests/test_umath.py

```diff
@@ -1271,6 +1271,29 @@ def test_heaviside(self):
         assert_equal(h, expected1.astype(np.float32))
 
 
+class TestSinc:
+
+    def test_special_cases(self):
+        assert_(ncu.sinc(0) == 1)
+        assert_(ncu.sinc(0 + 0j) == 1)
+        assert_(ncu.sinc(np.inf) == 0)
+        assert_(ncu.sinc(-np.inf) == 0)
+        assert_(np.isnan(ncu.sinc(np.nan)))
+        # Test a large zero
+        assert_(ncu.sinc(1e10) == 0)
+
+    def test_real(self):
+        x = np.linspace(-5, 5, 100)
+        assert_allclose(ncu.sinc(x), ncu.sin(np.pi*x)/(np.pi*x), atol=1e-16)
+
+    def test_complex(self):
+        x = np.linspace(-5, 5, 100)
+        y = np.linspace(-5, 5, 100)
+        x, y = np.meshgrid(x, y)
+        z = x + 1j*y
+        assert_allclose(ncu.sinc(z), ncu.sin(np.pi*z)/(np.pi*z))
```

Comment:
I wonder if we should add a test which checks for the improved relative precision compared to the previous method. Also we should have tests for all types, which is probably nicest by using pytest parameterization.